### PR TITLE
[SYCL][Matrix]Align the parameter Group of APIs {joint_matrix_store/joint_matrix_load} with matrix spec

### PR DESCRIPTION
**Pull request by wenhuiNi**
[SYCL][Matrix]Align the parameter Group of APIs {joint_matrix_store/joint_matrix_load} with matrix spec

### CODE REVIEW

**Review by YuriPlyakhin — COMMENTED** [review]
Please, update spec: https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc
Currently it doesn't have const specifier for group parameter.

**Comment by dkhaldi** [comment]
> I asked similar questions to @wenhuiNi in the chat:
> 
> * what is the context of this change or why we are adding this?
> * Since this changes the spec, it needs approval about the spec first.
> * the test does not expose this new addition

I got some context from @yubingex007-a11y. 
There is a divergence between the spec and the implementation in the sense that the Group argument should be passed by value, not reference. This will match the spec and also the other SYCL Group APIs like joint_reduce.
If the problem is with passing the get_sub_group() directly into a reference argument, see if passing Group argument by value solves the issue.

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_subgroup_xvalue_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#define TM 8
```
[the remark this answers is not in the document]
Comment (wenhuiNi) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_subgroup_xvalue_impl.hpp:1]: ok, I think E2E test is not necessary.  I will modify the https://github.com/intel/llvm/blob/sycl/sycl/test/matrix/matrix-bfloat16-test.cpp to check that the APIs can accept a xvalue or rvalue. Compile-time test is sufficient.

Thanks

**Comment by wenhuiNi** [comment]
Thanks @yubingex007-a11y 

@dkhaldi 
>If the problem is with passing the get_sub_group() directly into a reference argument, see if passing Group argument by value solves the issue.
Yes, It will solve this issue.

>Since this changes the spec, it needs approval about the spec first.
 I will modify the PR to be consistent with the spec, and I think it can solve my problem.

>the test does not expose this new addition
Yes, I will modify existing test Instead of introducing new tests

**Review by dkhaldi — APPROVED** [review]
LGTM
Thank you for detecting this inconsistency with the spec!

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

**Review by steffenlarsen — APPROVED** [review]

**Comment by steffenlarsen** [comment]
CUDA AWS is currently broken. Merging.

**Pull request merged**